Predictors/rnn.py

```diff
@@ -0,0 +1,172 @@
+from math import floor
+
+import numpy as np
+import tensorflow as tf
+from sklearn.base import BaseEstimator, ClassifierMixin
+
+
+class RNNClassifier(BaseEstimator, ClassifierMixin):
+    """
+    Generic RNN - can use ordinary RNN, LSTM or GRU
+    
+    Internal state is fed into a fully connected logistic regression layer and then softmax is applied to achieve classification
+       Basically the logistic regression layer provides a probability for each class.
+    
+    Based upon "Training a Sequence Classifier" on page 587 of "Hands on Machine Learning with Scikit-Learn and TensorFlow" by Aurelien Geron
+    """
+
+    def __init__(self, CellType=tf.contrib.rnn.GRUCell, n_neurons=100, learning_rate=0.001, n_outputs=4,
+                 n_out_neurons=10, n_inputs=50, n_steps=5, n_epochs=100, batch_size=100, **cell_args):
+        """
+        n_inputs = word encoding size
+        n_steps = amount of recursion e.g. window size
+        n_neurons = number of neurons in RNN
+        n_out_neurons = number of neurons in the fully connected output layer (= the number of classes)
+        n_outputs = number of outputs
+        """
+        self.CellType = CellType
+        self.n_neurons = n_neurons
+        self.learning_rate = learning_rate
+        self.n_outputs = n_outputs
+        self.n_out_neurons = n_out_neurons
+        self.n_inputs = n_inputs
+        self.n_steps = n_steps
+        self.n_epochs = n_epochs
+        self.batch_size = batch_size
+        self.cell_args = cell_args
+
+        self.construct_()
+
+    def construct_(self):
+        """
+        tensorflow construction stage
+        """
+
+        self.X = tf.placeholder(tf.float32, [None, self.n_steps, self.n_inputs])
+        # print("\n\nExpecting X.shape = {}, dtype = {}\n".format(self.X.shape, self.X.dtype))
+        # y = tf.placeholder(tf.int64, [None, self.n_outputs])
```

Comment:
is this commented out code needed in the future? If so, annotate what it is, otherwise I think it should be removed.